pyproject.toml

```diff
@@ -48,3 +48,62 @@ testpaths = [
 filterwarnings = [
     "ignore::DeprecationWarning",
 ]
+
+[tool.black]
+line-length = 88
+target-version = ['py38']
+
+[tool.isort]
+profile = "black"
+line_length = 88
+
+[tool.ruff]
+line-length = 88
+target-version = "py38"
+
+[tool.ruff.lint]
+select = [
+    "E",  # pycodestyle errors
+    "F",  # Pyflakes errors (undefined names, unused imports/variables)
+    "W",  # pycodestyle warnings
+    "B",  # flake8-bugbear (potential logic errors/style issues)
+    "C4",  # flake8-comprehensions (unnecessary list/set/dict comprehensions)
+    "SIM",  # flake8-simplify (opportunities to simplify code)
+    "UP",  # pyupgrade (syntax upgrades for newer Python versions)
+]
+ignore = [
+    "I",  # isort
+    "E501",  # Line too long, handled by black
+]
+
+[tool.ruff.lint.per-file-ignores]
+# temporal ignores, to be removed once we figure out how to refactor code
+"canopen/objectdictionary/__init__.py" = [
+    "F403",  # unable to detect undefined names
+    "F405",  # variable defined from star imports
+    "SIM115",  # context manager for opening files
+]
+"canopen/objectdictionary/eds.py" = [
+    "SIM115",  # context manager for opening files
+]
+"canopen/sdo/client.py" = [
+    "F403",  # unable to detect undefined names
+    "F405",  # variable defined from star imports
+    "SIM102",  # nested `if` statements
+]
+"canopen/sdo/server.py" = [
+    "F403",  # unable to detect undefined names
+    "F405",  # variable defined from star imports
+]
+"test/test_eds.py" = [
+    "SIM117",  # nested with statements
+]
+"test/test_local.py" = [
+    "SIM117",  # nested with statements
+]
```

Comment:
Definitely don't get started collecting exemptions like this.  Either fix it or accept that a warning will pop up unless somebody steps up to cleanly refactor and get rid of the root cause.